extensions/shared/track-gutenberg-search.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import { registerPlugin } from '@wordpress/plugins';
+
+/**
+ * Internal dependencies
+ */
+import analytics from '../../_inc/client/lib/analytics';
+
+const trackGutenbergSearch = event => {
+	const searchTerm = event.target.value;
+	const hasResults = document.getElementsByClassName( 'editor-inserter__no-results' ).length === 0;
+
+	if ( searchTerm === '' ) {
+		return;
+	}
+
+	analytics.tracks.recordEvent( 'jetpack_gutenberg_block_picker_search_term', {
+		searchTerm: searchTerm,
+	} );
+
+	if ( hasResults ) {
+		return;
+	}
+
+	// Create a separate event for search with no results to make it easier to filter by them
+	analytics.tracks.recordEvent( 'jetpack_gutenberg_block_picker_no_results', {
+		searchTerm: searchTerm,
+	} );
+};
+
+registerPlugin( 'track-no-search-results', {
+	render: () => {
+		document.onkeyup = event => {
+			if ( event.target.id.indexOf( 'editor-inserter__search' ) === 0 ) {
```

Comment:
In my tests, the search input has an id like `block-editor-inserter__search-0` or `block-editor-inserter__search-1` so this is failing to match

Reply:
Interesting. That's what I see on wpcom, where are you testing?

Reply:
This was happening because I was using the old version of Gutenberg. I've fixed it in 97c0931c955cfa43a0c503e486350170070679f6